bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/discovery/TouchWandUnitDiscoveryService.java

```diff
@@ -0,0 +1,217 @@
+                        if (!touchWandBridgeHandler.isAddSecondaryControllerUnits()) {
+                            if (!Arrays.asList(CONNECTIVITY_OPTIONS).contains(touchWandUnit.getConnectivity())) {
+                                continue;
+                            }
+                        }
+                        String type = touchWandUnit.getType();
+                        if (!Arrays.asList(SUPPORTED_TOUCHWAND_TYPES).contains(type)) {
+                            logger.debug("Unit discovery skipping unsupported unit type : {} ", type);
+                            continue;
+                        }
+                        switch (type) {
+                            case TYPE_WALLCONTROLLER:
+                                addDeviceDiscoveryResult(touchWandUnit, THING_TYPE_WALLCONTROLLER);
+                                break;
+                            case TYPE_SWITCH:
+                                addDeviceDiscoveryResult(touchWandUnit, THING_TYPE_SWITCH);
+                                notifyListeners(touchWandUnit);
+                                break;
+                            case TYPE_DIMMER:
+                                addDeviceDiscoveryResult(touchWandUnit, THING_TYPE_DIMMER);
+                                notifyListeners(touchWandUnit);
+                                break;
+                            case TYPE_SHUTTER:
+                                addDeviceDiscoveryResult(touchWandUnit, THING_TYPE_SHUTTER);
+                                break;
+                            default:
+                                continue;
+                        }
+                    }
+                } catch (JsonSyntaxException e) {
+                    logger.warn("Could not parse unit {}", e.getMessage());
+                }
+            }
+        } catch (JsonSyntaxException msg) {
+            logger.warn("Could not parse list units response {}", msg.getMessage());
+        }
+    }
+
+    private void notifyListeners(TouchWandUnitData touchWandUnit) {
+        /* sometimes current status received null , no point update listeners */
+        if (touchWandUnit.getCurrStatus() == null) {
+            return;
+        }
+
+        for (TouchWandUnitStatusUpdateListener listener : listeners) {
+            listener.onDataReceived(touchWandUnit);
+        }
+    }
+
+    @Override
+    protected synchronized void stopScan() {
+        removeOlderResults(getTimestampOfLastScan());
+        super.stopScan();
+    }
+
+    public void activate() {
+        super.activate(null);
+        removeOlderResults(new Date().getTime(), touchWandBridgeHandler.getThing().getUID());
+    }
+
+    @Override
+    public void deactivate() {
+        removeOlderResults(System.currentTimeMillis(), touchWandBridgeHandler.getThing().getUID());
+        super.deactivate();
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        if (scanningJob == null || scanningJob.isCancelled()) {
```

Comment:
See above. Store to a local variable to eliminate the compiler warning.

Reply:
I been try it and the scanning job failed , for now I will commit the change but doing some more testing to verify it is a valid solution.